src/gatsby-node.js

```diff
@@ -302,6 +305,40 @@ exports.sourceNodes = async (
         );
       }
     };
+    ws.onclose = function() {
+      reconnectingWebSocket(ws, previousDelay);
+    };
+    ws.onerror = function(err) {
+      reporter.error(err.stack);
+    };
+
+    ws.on('pong', () => {
+      count--;
+    });
+
+    const intervalId = setInterval(() => {
+      if (ws.readyState == 1) {
+        ws.ping('heartBeat');
+        count++;
+      }
+      if (count == 2) {
+        count = 0;
+        reconnectingWebSocket(ws, previousDelay);
+      }
+    }, 60000);
+  };
+
+  const reconnectingWebSocket = function(ws, previousDelay) {
+    previousDelay = Math.min(60, previousDelay * (2 - Math.random()));
+    timeoutId = setTimeout(() => {
+      if (ws.readyState == 3) {
+        webSocketStart();
```

Comment:
previousDelay should be passed as argument to webSocketStart so that it can increase by time